### PR TITLE
Task ys/j134 add select podcast genre functionality

### PR DESCRIPTION
**Pull request by yaroslav207**

### CODE REVIEW

File: backend/src/data/migrations/20210813151449_add_genre_id_to_podcasts.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const TABLE_NAME = 'podcasts';
+const COLUMN_NAME = 'genre_id';
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.schema.table(TABLE_NAME, (table) => {
+    table
+      .integer(COLUMN_NAME)
+      .references('id')
+      .inTable('genres')
+      .notNullable();
```
Comment (ghost) [inline backend/src/data/migrations/20210813151449_add_genre_id_to_podcasts.ts]:
```suggestion
      .default(null);
```

File: backend/src/services/genre/genre.service.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import {
+  Genre as TGenre,
+} from '~/common/types/types';
+
```
Comment (ghost) [inline backend/src/services/genre/genre.service.ts]:
```suggestion
```

File: backend/src/validation-schemas/podcast/create-podcast/create-podcast.validation-schema.ts [hunk 3]
```diff
@@ -16,6 +16,12 @@ const podcastCreate = podcast.keys({
     }),
     Joi.any().equal(null),
   ],
+  [PodcastPayloadKey.GENRE_ID]: Joi.number()
+    .integer()
+    .required()
```
Comment (ghost) [inline backend/src/validation-schemas/podcast/create-podcast/create-podcast.validation-schema.ts]: it isn't require

File: backend/src/validation-schemas/podcast/edit-podcast/edit-podcast.validation-schema.ts [hunk 4]
```diff
@@ -20,6 +20,12 @@ const podcastEdit = podcast.keys({
     }),
     Joi.any().equal(null),
   ],
+  [PodcastPayloadKey.GENRE_ID]: Joi.number()
```
Comment (ghost) [inline backend/src/validation-schemas/podcast/edit-podcast/edit-podcast.validation-schema.ts:23]: it isn't require

File: backend/src/data/models/podcast/podcast.model.ts [hunk 5]
```diff
@@ -19,6 +19,8 @@ class Podcast extends Abstract {
 
   [PodcastDTOKey.TYPE]: PodcastType;
 
+  [PodcastDTOKey.GENRE_ID]: number;
```
Comment (ghost) [inline backend/src/data/models/podcast/podcast.model.ts]:
```suggestion
  [PodcastDTOKey.GENRE_ID]: number | null;
```

File: frontend/src/components/configurate-podcast/components/configurate-podcast-form/common/constants.ts [hunk 6]
```diff
@@ -6,6 +6,7 @@ const DEFAULT_PODCAST_PAYLOAD: PodcastFormPayload = {
   [PodcastPayloadKey.DESCRIPTION]: '',
   [PodcastPayloadKey.IMAGE]: null,
   [PodcastPayloadKey.TYPE]: PodcastType.PUBLIC,
+  [PodcastPayloadKey.GENRE]: '',
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-form/common/constants.ts]:
```suggestion
  [PodcastPayloadKey.GENRE]: null,
```

File: frontend/src/store/genre/reducer.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { createReducer } from '@reduxjs/toolkit';
```
Comment (ghost) [inline frontend/src/store/genre/reducer.ts]: it should be a new reducer
add this actions & reducer keys where you need

File: frontend/src/store/root-reducer.ts [hunk 8]
```diff
@@ -7,6 +7,7 @@ import { reducer as episode } from './episode/reducer';
 import { reducer as podcast } from './podcast/reducer';
 import { reducer as record } from './record/reducer';
 import { reducer as userProfile } from './user-profile/reducer';
+import { reducer as genre } from './genre/reducer';
```
Comment (ghost) [inline frontend/src/store/root-reducer.ts]:
```suggestion
```

File: frontend/src/validation-schemas/podcast/create-podcast/create-podcast.validation-schema.ts [hunk 9]
```diff
@@ -22,6 +22,11 @@ const podcastCreate = podcast.keys({
     .messages({
       'file.invalidExtension': PodcastValidationMessage.FILE_EXTENSION_FORMAT,
     }),
+  [PodcastPayloadKey.GENRE]: Joi.string()
+    .required()
```
Comment (ghost) [inline frontend/src/validation-schemas/podcast/create-podcast/create-podcast.validation-schema.ts]: nullable

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/configurate-podcast/helpers/map-podcast-to-form-payload/map-podcast-to-form-payload.ts [hunk 10]
```diff
@@ -5,6 +5,7 @@ const mapPodcastToFormPayload = (podcast: Podcast): PodcastFormPayload => ({
   description: podcast.description,
   image: null,
   type: podcast.type,
+  genre: (podcast.genreId === null) ? null : podcast.genreId.toString(),
```
Comment (what1s1ove) [inline frontend/src/components/configurate-podcast/helpers/map-podcast-to-form-payload/map-podcast-to-form-payload.ts]:
```suggestion
  genre: podcast.genreId?.toString() ?? null,
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**